### ISSUE TITLE
Bad instructions to get email address

### ISSUE THREAD

**Issue by Morriz** [strongloop/loopback-component-passport#147]
In the readme of this repo I read this:

_If more info is required from the Facebook profile such as email, it could still be obtained. In node_modules\passport-facebook\lib\strategy.js, replace:_

You are asking people to modify code in node_modules. Which is very bad. Is there really no functionality to do this otherwise?

**Comment by superkhau** [comment]
@loay I recall you working on parts of this update. Do you know if this is a requirement of the module?

**Comment by alexpls** [comment]
From what I've found there is a `profileToUser` function which is exposed and is overridable as part of the strategy's configuration. There doesn't seem to be much documentation around this though, I just found out about it by reading the code: https://github.com/strongloop/loopback-component-passport/blob/master/lib/models/user-identity.js#L122

@Morriz Overriding that function as part of your strategy's config should solve the issue and not require you to override anything in the `node_modules` folder.

**Comment by Morriz** [comment]
Tnx. Saw it. Bad way to deal with this tho. Did you see what goes on in that method?

**Comment by yotamsha** [comment]
I had to override this function as well. But it seems like a bad idea since there is some functionality there which I had to repeat, like the dummy password hash generatation. (plus I couldn't find any clear documentation about this)

**Comment by clockworkgr** [comment]
Yep, you do not have change to override strategy.js.

This issue was originally referenced here https://github.com/strongloop/loopback-example-passport/issues/50 when FB changed their default response.

Somehow a recommendation to override a function on an external package such as passport-facebook  is considered a cleaner solution than a configuration setting to ask for the facebook profile fields you need.

Go figure...

TLDR; user profileFields in providers.json

**Comment by loay** [comment]
Thanks for pointing that out. 
- readme file will be updated shortly. 
- A pr is created to fix that issue is `loopback-example-passport` and it will be merged shortly. 
- Documentation is updated: http://loopback.io/doc/en/lb2/Configuring-providers.json.html#example-providersjson

**Comment by loay** [comment]
Both PR merged. Closing issue.
